edb/schema/objtypes.py

```diff
@@ -640,9 +640,12 @@ def _alter_finalize(
                 # them in the schema to produce the correct as_alter_delta.
                 nschema = _delete_to_delist(delete, schema)
 
-                nschema, nunion = utils.get_union_type(
+                types = utils.simplify_union_types(
+                    schema, union.get_union_of(schema).objects(schema)
+                )
```

Comment:
Do we want this simplification? It seems like probably not?

Also the surrounding code seems bizarre and makes no sense to me, and so of course I wrote it.